openedx/stanford/cms/djangoapps/contentstore/static/js/factories/bulkupdate.js

```diff
@@ -0,0 +1,147 @@
+            SHOW_ANSWER_OPTIONS = [],
+            VALIDATION_ERROR_MESSAGES = {
+                maxAttempts: gettext('MaxAttempts must be a non-negative integer. Please enter a different value.'),
+                showAnswer: gettext('Not a valid value for showAnswer. Please enter a different value.')
+            };
+
+        // Fill SHOW_ANSWER_OPTIONS with option values from the HTML
+        $('#show-answer option').each(function() {
+            SHOW_ANSWER_OPTIONS.push($(this).val());
+        });
+
+        /**
+         * Hides all error and success messages
+         */
+        function clearFeedbackDisplay() {
+            $('.error-message-text').text('');
+            $errorMessage.addClass('is-hidden');
+            $successMessage.addClass('is-hidden');
+        }
+
+        /**
+         * Gets setting values from HTML
+         * @return {dict} Setting values the user wants changed, empty string
+         *                value if they don't want to change that setting
+         */
+        function getSettingsData() {
+            var data = {};
+            if ($applyMaxAttempts.is(':checked')) {
+                data.maxAttempts = parseInt($maxAttempts.val(),10);
+            } else {
+                data.maxAttempts = '';
+            }
+            if ($applyShowAnswer.is(':checked')) {
+                data.showAnswer = $showAnswer.val();
+            } else {
+                data.showAnswer = '';
+            }
+            return data;
+        }
+
+        /**
+         * Validates setting values after being processed by getSettingsData()
+         * @param {dict} settings data from getSettingsData()
+         * @return {bool} false if no settings changed or settings have invalid value
+         */
+        function validateData(data) {
+            var errors = [],
+                errorString = '',
+                maxAttempts = data.maxAttempts,
+                showAnswer = data.showAnswer;
+
+            if (!maxAttempts && maxAttempts !== 0 && !showAnswer) {
+                return false;
+            }
+            if (maxAttempts && maxAttempts < 0) {
+                errors.push(VALIDATION_ERROR_MESSAGES.maxAttempts);
+            }
+            if (showAnswer && SHOW_ANSWER_OPTIONS.indexOf(showAnswer) < 0) {
+                errors.push(VALIDATION_ERROR_MESSAGES.showAnswer);
+            }
+            if (errors.length > 0) {
+                errorString = errors.join(' ');
+                $('.error-message-text').text(errorString).show();
+                return false;
+            }
+            return true;
+        }
+
+        /**
+         * Displays error message on request error
+         */
+        function onError(response) {
+            var errMsg;
+            var serverMsg;
+            try {
+                serverMsg = $.parseJSON(response.responseText) || {};
+                errMsg = serverMsg.ErrMsg;
+            } catch (e) {
+                errMsg = '';
```

Comment:
Should we display the exception text as the error message?

It seems a little weird that our error-handling in our error-handling doesn't display an error :)